From a63c5b109f03d0119822592455ba9c2bc1f709d9 Mon Sep 17 00:00:00 2001
From: Ken VanDine <ken.vandine@canonical.com>
Date: Fri, 19 Nov 2010 13:37:01 -0500
Subject: [PATCH] Include <package/> tag in generated GIR

Specify pkg-config name when generating GIR files, used for mapping GIR
name to pkg-config names

https://bugzilla.gnome.org/show_bug.cgi?id=635287
---
 gdk/Makefile.am |    2 ++
 gtk/Makefile.am |    1 +
 2 files changed, 3 insertions(+), 0 deletions(-)

Index: gtk+-2.24.7/gdk/Makefile.am
===================================================================
--- gtk+-2.24.7.orig/gdk/Makefile.am	2011-08-16 04:30:51.000000000 +0200
+++ gtk+-2.24.7/gdk/Makefile.am	2011-10-18 19:11:08.000000000 +0200
@@ -3,6 +3,7 @@
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = \
+   --warn-all \
    --add-include-path=../gdk
 INTROSPECTION_COMPILER_ARGS = \
    --includedir=$(srcdir) \
@@ -198,10 +199,11 @@
         gdkenumtypes.h
 
 Gdk-2.0.gir: $(gdktargetlib) Makefile
-Gdk_2_0_gir_SCANNERFLAGS = --warn-all
+Gdk_2_0_gir_SCANNERFLAGS = --c-include="gdk/gdk.h"
 Gdk_2_0_gir_INCLUDES = Gio-2.0 GdkPixbuf-2.0 Pango-1.0 cairo-1.0
 Gdk_2_0_gir_LIBS = $(gdktargetlib)
 Gdk_2_0_gir_FILES = $(introspection_files)
+Gdk_2_0_gir_PACKAGES = gdk-2.0
 Gdk_2_0_gir_CFLAGS = $(INCLUDES)
 INTROSPECTION_GIRS += Gdk-2.0.gir
 
@@ -243,6 +244,7 @@
 GdkX11_2_0_gir_INCLUDES = Gio-2.0 Gdk-2.0 GdkPixbuf-2.0 Pango-1.0 xlib-2.0
 GdkX11_2_0_gir_LIBS = $(gdktargetlib)
 GdkX11_2_0_gir_FILES = $(x11_introspection_files)
+GdkX11_2_0_gir_EXPORT_PACKAGES = gdk-x11-2.0
 GdkX11_2_0_gir_CFLAGS = $(INCLUDES) -L$(top_builddir)/gdk
 INTROSPECTION_GIRS += GdkX11-2.0.gir
 
Index: gtk+-2.24.7/gtk/Makefile.am
===================================================================
--- gtk+-2.24.7.orig/gtk/Makefile.am	2011-10-18 00:38:28.000000000 +0200
+++ gtk+-2.24.7/gtk/Makefile.am	2011-10-18 19:11:08.000000000 +0200
@@ -3,6 +3,7 @@
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = \
+	--warn-all \
 	--add-include-path=../gdk
 INTROSPECTION_COMPILER_ARGS = \
 	--includedir=$(srcdir) \
@@ -1010,11 +1011,11 @@
     gtktypebuiltins.c
 
 Gtk-2.0.gir: $(INTROSPECTION_SCANNER) $(gtktargetlib) $(top_builddir)/gdk/Gdk-2.0.gir Makefile
-Gtk_2_0_gir_SCANNERFLAGS = --warn-all --add-include-path=$(top_builddir)/gdk
+Gtk_2_0_gir_SCANNERFLAGS = --include-uninstalled=$(top_builddir)/gdk/Gdk-2.0.gir
 if USE_X11
 Gtk_2_0_gir_SCANNERFLAGS += --add-include-path=$(top_builddir)/gdk/x11
 endif
-Gtk_2_0_gir_INCLUDES = Atk-1.0 Gdk-2.0
+Gtk_2_0_gir_INCLUDES = Atk-1.0
 Gtk_2_0_gir_CFLAGS = \
 		$(INCLUDES) \
 		-UGDK_DISABLE_DEPRECATED \
@@ -1022,6 +1023,7 @@
 		-DGTK_TEXT_USE_INTERNAL_UNSUPPORTED_API
 Gtk_2_0_gir_LIBS = $(gtktargetlib)
 Gtk_2_0_gir_FILES = $(addprefix $(srcdir)/, $(introspection_files))
+Gtk_2_0_gir_PACKAGES = gtk-2.0
 INTROSPECTION_GIRS += Gtk-2.0.gir
 
 girdir = $(datadir)/gir-1.0
